frontend/src/lib/utils/geometry.ts

```diff
@@ -18,51 +23,51 @@ export type Rect = {
 export const ORIGIN = Object.freeze({ x: 0, y: 0 });
 export const MANHATTAN_LENGTH = 13.11;
 
-export function pointerEventToPoint(event: PointerEvent): Point {
+export function pointerEventToPoint(event: PointerEvent): Point2D {
     return { x: event.pageX, y: event.pageY };
 }
 
-export function vectorLength(vector: Point): number {
+export function vectorLength(vector: Vector2D): number {
     return Math.sqrt(Math.pow(vector.x, 2) + Math.pow(vector.y, 2));
 }
 
-export function pointDistance(point1: Point, point2: Point): number {
+export function pointDistance(point1: Point2D, point2: Point2D): number {
     return Math.sqrt(Math.pow(point1.x - point2.x, 2) + Math.pow(point1.y - point2.y, 2));
 }
 
-export function pointDifference(point1: Point, point2: Point): Point {
-    return { x: point1.x - point2.x, y: point1.y - point2.y };
+export function vectorDifference(vector1: Vector2D, vector2: Vector2D): Vector2D {
```

Comment:
Change to `pointDifference(point1: Point2D, point2: Point2D): Point2D`, as it is only used with points so far?